components/homme/src/share/namelist_mod.F90

```diff
@@ -197,9 +192,9 @@ module namelist_mod
   ! read in the namelists...
   ! ============================================
 
-#ifdef CAM
+#if defined(CAM) || defined(SCREAM)
   subroutine readnl(par, NLFileName)
-    use units, only : getunit, freeunit
+    use shr_file_mod,      only: getunit=>shr_file_getUnit, freeunit=>shr_file_freeUnit
```

Comment:
This allows to build scream without pulling in quite a few of the EAM modules. The eam module `units` was literally just wrapping the corresponding subroutines in `shr_file_mod`, without adding any logic whatsoever.